src/components/Paystack/ShowPaystack.vue

```diff
@@ -0,0 +1,221 @@
+		<div class="modal-dialog">
+			<form
+				action="javascript:"
+				class="modal-content"				
+				id="paystackForm"
+			>
+				<div class="modal-header ml-5 py-3 text-center">
+					<h6 class="modal-title my-modal text-center w-75">
+						Get Paystack Auth Code
+					</h6>
+					<a
+						aria-label="Close"
+						class="close py-1"
+						data-dismiss="modal"
+						href="javascript:"
+					>
+						<span class="modal-close text-danger"
+							><i class="fas fa-times"></i
+						></span>
+					</a>
+				</div>
+				<div class="modal-body" id="index" v-if="customer || order">
+					<div class="form-group clearfix">
+						<div class="clearfix">
+							<div class="form-group">
+								<label>Customer's Name.</label>
+								<span class="font-weight-bold mx-2">{{ order.customer_name }}</span>
+
+								
+							</div>
+
+							<div class="form-group">
+								<label>Customer's Phone Number.</label>
+								<span class="font-weight-bold mx-2">{{ order.customer_phone }}</span>
+
+								
+							</div>
+
+							<div class="form-group">
+								<label>Order Number: </label>
+								<span class="font-weight-bold mx-2">{{ orderNumber }}</span>
+							</div>
+
+							<div class="form-group">
+								<label>Amount to charge: </label>
+								<span class="font-weight-bold mx-2">₦100</span>
+							</div>
+						</div>
+					</div>
+				</div>
+				<div class="modal-footer d-flex justify-content-end">
+					<paystack
+						:amount="10000"
+						:email="customer ? customer.email : ''"
+						:paystackkey="paystackkey"
+						:reference="reference"
+						:callback="processPaymentPayStackPayment"
+						:close="closePayStackModal"
+						class="btn bg-default"
+						>Proceed</paystack
+					>
+
+					<button class="btn btn-danger rounded-1 mx-2" data-dismiss="modal" href="javascript:">
+						Cancel
+					</button>
+				</div>
+			</form>
+		</div>
+	</div>
+</template>
+
+<script>
+	import Vue from 'vue';
+	import { mapGetters } from 'vuex';
+	import { post } from '../../utilities/api';
+	import Flash from '../../utilities/flash';
+	import { Message } from '../../utilities/sms';
+	import { EventBus } from '../../utilities/event-bus';
+	import paystack from 'vue-paystack';
+
+	export default {
+		components: {
+			paystack,
+		},
+
+		data() {
+			return {
+				order: null,
+				value: null,
+				DSAId: null,
+				DSAName: null,
+				message: null,
+				customer: null,
+				messageType: 'custom',
+				showChangeCustomerManagerModal: true,
+				verifyPaymentUrl: `https://api.paystack.co/transaction/verify/`,
+				paystack_auth_code_url: '/api/pay_stack_auth_code',
+				paystackkey: process.env.VUE_APP_PAYSTACK_KEY || '',
+				paystack_secret_key: process.env.VUE_APP_PAYSTACK_SECRET_KEY || '',
+				authorization_code: null,
+				paystackReference: null,
+			};
+		},
+
+		computed: {
+			reference() {
+				let text = '';
+				let possible =
+					'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789' +
```

Comment:
What are we doing here?

Reply:
Generating a random reference for paystack. it is needed before loading widget